### PR TITLE
Templatize concourse pipeline

### PR DESCRIPTION
**Pull request by jimmykarily**
## Description
Make it possible to easily deploy a copy of the concourse pipeline pointing to different git repositories and S3 buckets.

## Motivation and Context
People contributing to this project may want to:

- Deploy their own version of the pipeline, consuming their own hardware resources to speed up development.
- Deploy a copy of the pipeline in order to make changes to the pipeline itself before opening a PR.

## How Has This Been Tested?
Run the "fly" wrapper script with no config.yaml in place. It deploys the same pipeline that is currently deployed. When run with a config.yaml, it makes the relevant adjustments.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code has security implications.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: .concourse/README.md [hunk 1]
```diff
@@ -48,8 +48,14 @@ point to your new EKCP API endpoint.
 ## Deploy on Concourse
 
 If you wish to deploy the pipeline on Concourse, run the following
-command:
+command and use the `fly` script in this directory:
 
 ```
-fly -t target set-pipeline -c pipeline.yaml -p kubecf
+./fly -t target set-pipeline -c pipeline.yaml -p kubecf
```
Comment (bikramnehra) [inline .concourse/README.md]: Can't we have `fly` binary in `$PATH` to maintain status quo.?

File: .concourse/config.yaml.sample [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# All null (~) values have defaults in the pipeline
```
Comment (bikramnehra) [inline .concourse/config.yaml.sample]: How can we absolutely make sure that there is no/minimal possibility for secret leaks? As the pipeline might be having public access.

File: .concourse/README.md [hunk 3]
```diff
@@ -48,8 +48,14 @@ point to your new EKCP API endpoint.
...
 ```
+
+## Pipeline development
```
Comment (bikramnehra) [inline .concourse/README.md]: Cool. Good to have an example :+1:

File: .concourse/README.md (hunk 1 above)
Comment (jimmykarily) [reply .concourse/README.md]: This `fly` here is the script we include in this PR. It's wrapping fly to render the template. See the fly script below.

File: .concourse/config.yaml.sample (hunk 2 above)
Comment (jimmykarily) [reply .concourse/config.yaml.sample]: We removed the secrets from the values here. You are right, we don't need to templetize them. If someone is deploying on a different concourse instance, they will simpy add their own secrets in their secret storage solution.

**Comment by jimmykarily** [comment]
@bikramnehra we addressed your comments and got @mudler approval. Merging.

**Pull request merged**